utils/common/src/main/java/org/apache/brooklyn/util/exceptions/Exceptions.java

```diff
@@ -46,12 +47,19 @@
 
 public class Exceptions {
 
-    private static final List<Class<? extends Throwable>> BORING_THROWABLE_SUPERTYPES = ImmutableList.<Class<? extends Throwable>>of(
-        ExecutionException.class, InvocationTargetException.class, PropagatedRuntimeException.class, UndeclaredThrowableException.class);
+    private static final List<Class<? extends Throwable>> ALWAYS_BORING_THROWABLE_SUPERTYPES = ImmutableList.<Class<? extends Throwable>>of(
+        ExecutionException.class, InvocationTargetException.class, UndeclaredThrowableException.class);
```

Comment:
A concrete example of really bad behaviour would be if `task.get()` finds an exception (wrapped in an `ExecutionException`), unwraps it, and then re-throws this "nicer" exception. At that point, we've lost the stacktrace that got us to the code which called `task.get()`, and only have the stacktrace of the other thread. Also, if the other thread's failure was an `InterruptedException` which we then re-throw, then we could end up interrupting our own thread even though we weren't supposed to!

I'm not saying we're doing this anywhere (I hope not!), but we need to be very careful about how we use `isBoring(t)` and `getFirstInteresting(t)` - both for logging and for propagating.